search-app/src/cmr/search/services/query_execution/has_granules_created_at_feature.clj

```diff
@@ -0,0 +1,76 @@
+   [cmr.common-app.services.search.params :as common-params]
+   [cmr.common-app.services.search.query-execution :as query-execution]
+   [cmr.common-app.services.search.query-model :as query-model]
+   [cmr.common-app.services.search.query-to-elastic :as query-to-elastic]
+   [cmr.search.services.parameters.conversion :as parameters]
+   [cmr.search.services.query-walkers.update-field-resolver :as update-field-resolver]))
+
+(def query-aggregation-size
+  "Page size for query aggregations. This should be large enough
+  to contain all collections to allow searching for collections
+  from granules."
+  50000)
+
+(defn has-granules-created-at-query
+  "Generates a query to find any collection-concept-ids for granules that were created in the
+  provided time range(s)."
+  [condition]
+  (query-model/query
+   {:concept-type :granule
+    :condition condition
+    :page-size 0
+    :result-format :query-specified
+    :result-fields []
+    :aggregations {:collections
+                   {:terms {:size query-aggregation-size
+                            :field (query-to-elastic/query-field->elastic-field
+                                    :collection-concept-id :granule)}}}}))
+
+(defn- generate-granule-query
+  "Creates a query to get the granules based on the provided collection query."
+  [query]
+  (as-> query query
+        ;; Change all collection field names to granule field names in the query
+        (reduce (fn [upd-query [orig-field-key new-field-key]]
+                  (update-field-resolver/rename-field upd-query orig-field-key new-field-key))
+                query
+                parameters/collection-to-granule-params)
+        ;; Remove any collection specific fields from the query
+        (reduce (fn [upd-query field-to-remove]
+                  (update-field-resolver/remove-field upd-query field-to-remove))
+                query
+                parameters/collection-only-params)
+        (has-granules-created-at-query (:condition query))))
+
+(defn- generate-collection-query-condition
+  "Takes the original query and the concept IDs from the granules query and constructs a new
+  collection query condition to execute. If there are not any concept-ids we should return a
+  match none condition."
+  [context query concept-ids]
+  (let [concept-ids-condition (when (seq concept-ids)
+                                (common-params/parameter->condition context :collection :concept-id
+                                                                    concept-ids nil))
+        ;; Remove the has-granules-created-at query condition
+        query (update-field-resolver/remove-field query :has-granules-created-at)]
+    (if (seq concept-ids)
+      (group-query-conditions/and-conds [(:condition query) concept-ids-condition])
+      query-model/match-none)))
+
+(defmethod query-execution/pre-process-query-result-feature :has-granules-created-at
+  [context query feature]
+  (let [granule-query (generate-granule-query query)
```

Comment:
If i am reading the code correctly, this searching sequence is not very straightforward and could potentially be a lot more expensive with certain search params. I have thought a more straightforward search sequence would be to filter the collections down first based on the search params, then use the filtered down collection identifiers to do the granule facet search.